Keep getting errors with the makefile targets

the files to touch seem to be complaining...
eg. make sources gives:

```make: *** No rule to make target `.make.process_sources', needed by `process_sources'.  Stop.```

--- a/hooks/post_gen_project.py
+++ b/hooks/post_gen_project.py
@@ -7,6 +7,14 @@ from cookiecutter.config import get_user_config
 logger = logging.getLogger(__name__)
 
 def copy_cookiecutter_resume(template_name='cookiecutter-easydata'):
+    """Make a copy of the cookiecutter replay file in the generated project.
+
+    By default, cookiecutter creates a replay directory in a user's ~/.cookiecutter
+    directory. This hook copies that file to the generated project. This can't
+    currently be used by cookiecutter, but it's nice to at least have this
+    information saved for now.
+    """
+
     config_obj = get_user_config()
     config_dir = pathlib.Path(config_obj['replay_dir'])
 
@@ -16,11 +24,10 @@ def copy_cookiecutter_resume(template_name='cookiecutter-easydata'):
     logger.debug(f"Reading cookiecutter replay data from {src_path}")
     with open(src_path) as f:
         cookiecutter_opts = json.load(f)
-        
+
     logger.info(f"Dumping cookiecutter replay info to {dst_path}")
     with open(dst_path, 'w') as fw:
         json.dump(cookiecutter_opts.get('cookiecutter', {}), fw, sort_keys=True, indent=4)
 
 if __name__ == '__main__':
     copy_cookiecutter_resume()
-    

--- a/{{ cookiecutter.repo_name }}/Makefile
+++ b/{{ cookiecutter.repo_name }}/Makefile
@@ -14,8 +14,13 @@ include Makefile.envs
 #
 
 .PHONY: requirements
+
 requirements: update_environment
 	@echo "WARNING: 'make requirements' is deprecated. Use 'make update_environment'"
+
+.PHONY: unfinished
+unfinished:
+	@echo "WARNING: this target is unfinished and may be removed or changed dramatically in future releases"
 
 #
 # COMMANDS                                                                      #
@@ -52,17 +57,17 @@ transform_data: .make.transform_data
 
 .PHONY: train
 ## train / fit / build models
-train: models/model_list.json
+train: unfinished models/model_list.json
 	$(PYTHON_INTERPRETER) -m $(MODULE_NAME).models.train_models model_list.json
 
 .PHONY: predict
 ## predict / transform / run experiments
-predict: models/predict_list.json
+predict: unfinished models/predict_list.json
 	$(PYTHON_INTERPRETER) -m $(MODULE_NAME).models.predict_model predict_list.json
 
 .PHONY: analysis
 ## Convert predictions / transforms / experiments into output data
-analysis: reports/analysis_list.json
+analysis: unfinished reports/analysis_list.json
 	$(PYTHON_INTERPRETER) -m $(MODULE_NAME).analysis.run_analysis analysis_list.json
 
 .PHONY: clean

--- a/{{ cookiecutter.repo_name }}/Makefile
+++ b/{{ cookiecutter.repo_name }}/Makefile
@@ -15,7 +15,7 @@ include Makefile.envs
 
 .PHONY: requirements
 requirements: update_environment
-	@echo "WARNING: `make requirements` is deprecated. Use `make update_environment`"
+	@echo "WARNING: 'make requirements' is deprecated. Use 'make update_environment'"
 
 #
 # COMMANDS                                                                      #

--- a/{{ cookiecutter.repo_name }}/Makefile
+++ b/{{ cookiecutter.repo_name }}/Makefile
@@ -35,23 +35,31 @@ data: transform_data
 sources: process_sources
 
 .PHONY: fetch_sources
-fetch_sources: .make.fetch_sources $(RAW_DATASOURCES)
+fetch_sources: .make.fetch_sources
+
+.make.fetch_sources:
 	$(PYTHON_INTERPRETER) -m $(MODULE_NAME).data.make_dataset fetch
 	touch .make.fetch_sources
 
 .PHONY: unpack_sources
 unpack_sources: .make.unpack_sources
+
+.make.unpack_sources: .make.fetch_sources
 	$(PYTHON_INTERPRETER) -m $(MODULE_NAME).data.make_dataset unpack
 	touch .make.fetch_sources
 
 .PHONY: process_sources
 process_sources: .make.process_sources
+
+.make.process_sources: .make.unpack_sources
 	$(PYTHON_INTERPRETER) -m $(MODULE_NAME).data.make_dataset process
 	touch .make.process_sources
 
 .PHONY: transform_data
 ## Apply Transformations to produce fully processed Datsets
 transform_data: .make.transform_data
+
+.make.transform_data: .make.process_sources
 	$(PYTHON_INTERPRETER) -m $(MODULE_NAME).data.apply_transforms
 	touch .make.transform_data
 

--- a/{{ cookiecutter.repo_name }}/{{ cookiecutter.module_name }}/data/datasets.py
+++ b/{{ cookiecutter.repo_name }}/{{ cookiecutter.module_name }}/data/datasets.py
@@ -9,7 +9,7 @@ from functools import partial
 
 from ..paths import processed_data_path, data_path, raw_data_path, interim_data_path, catalog_path
 from ..logging import logger
-from .fetch import fetch_file, unpack, get_dataset_filename
+from .fetch import fetch_file,  get_dataset_filename, hash_file, unpack
 from .utils import partial_call_signature, serialize_partial, deserialize_partial, process_dataset_default
 from ..utils import load_json, save_json
 

